### ISSUE TITLE
Code generator shouldn't generate JPA Column precision on non-decimal types

### ISSUE THREAD

**Issue by rayman245** [jOOQ/jOOQ#14132]
### Expected behavior

When generating pojo, float values (Double) are supposed to be annotated with precision of 13 instead of 53. Hibernate has changed this since Hibernate 5, this is currently breaking in Spring Boot 3.

https://hibernate.atlassian.net/browse/HHH-15623

### Actual behavior

Error from Hibernate, as it tries to convert it to 177bit signature.

### Steps to reproduce the problem

More details here:
https://hibernate.atlassian.net/browse/HHH-15623
https://github.com/spring-projects/spring-data-jpa/issues/2684#issuecomment-1290185932

### jOOQ Version

3.17.4

### Database product and version

Postgres

### Java Version

19

**Comment by lukaseder** [comment]
Thanks for your report. I'm sorry, I'm not sure I understand. What is the *jOOQ specific* problem here? Can you please provide a reproducer? In the issue template, we've suggested providing an MCVE (template here: https://github.com/jOOQ/jOOQ-mcve). Simpler reproduction instructions might work too, but I definitely need to know how to reproduce any problem.

**Comment by rayman245** [comment]
Hi @lukaseder I am getting this error when trying to generate DDL from Entity, only occurring for Double values with precision 53.

```
org.hibernate.tool.schema.spi.CommandAcceptanceException: Error executing DDL "create table <table> (value float(177))" via JDBC Statement
.
.
org.springframework.boot.web.servlet.context.ServletWebServerApplicationContext.refresh(ServletWebServerApplicationContext.java:146) ~[spring-boot-3.0.0-RC1.jar:3.0.0-RC1]
	at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:730) ~[spring-boot-3.0.0-RC1.jar:3.0.0-RC1]
	at org.springframework.boot.SpringApplication.refreshContext(SpringApplication.java:432) ~[spring-boot-3.0.0-RC1.jar:3.0.0-RC1]
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:308) ~[spring-boot-3.0.0-RC1.jar:3.0.0-RC1]
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1302) ~[spring-boot-3.0.0-RC1.jar:3.0.0-RC1]
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1291) ~[spring-boot-3.0.0-RC1.jar:3.0.0-RC1]
	at com.Application.main(Application.java:8) ~[classes/:na]
Caused by: org.postgresql.util.PSQLException: ERROR: precision for type float must be less than 54 bits
```
As shown here the generated entity (From JOOQ) has the correct precision 53,image

![image](https://user-images.githubusercontent.com/11938827/197923184-2b97febb-d105-4ca8-899b-b15204baf6f1.png)

All double precision seems to be translating to 177 bits instead of using the defined 53 bits. We are trying to upgrade from Spring-Boot v2 to v3. This was working fine in Spring Boot v2.

I have raised this issue to Hibernate, seems that in Hibernate 6 the mapping of 53 no longer works. More info on that from Christians comment here https://hibernate.atlassian.net/browse/HHH-15623

**Comment by lukaseder** [comment]
Thanks for your feedback. In our integration tests, this doesn't seem to be a problem. But we're not using Hibernate 6 yet in the `jooq-meta-extensions-hibernate` module.

But what does this have to do with jOOQ? Hibernate used to accept that precision and now rejects it. So, I guess you just remove it?

**Comment by lukaseder** [comment]
Just to be sure. Here you say:

> Hi @lukaseder I am getting this error when trying to generate DDL from Entity, only occurring for Double values with precision 53.

But on https://hibernate.atlassian.net/browse/HHH-15623, you say:

> As shown here the generated entity (From JOOQ) has the correct precision 53,image

What is it, then? Are you generating entities using jOOQ or are you using jOOQ to generate code from entities?

**Comment by lukaseder** [comment]
I see, we do generate that on POJOs. Guess we can remove it.

**Comment by lukaseder** [comment]
There's no formal information yet about whether a `jOOQ-meta` `DataTypeDefinition` has precision/length/scale (see https://github.com/jOOQ/jOOQ/issues/12048). So, this particular precision is an artifact of H2's code generation, which reports double precisions.

But we've worked around this already in #6411 for the `TableImpl` type declaration, so it might be possible to do this here as well.

**Comment by lukaseder** [comment]
While I'm fixing this, can you explain why you're using jOOQ's `<jpaAnnotations/>` flag? You're expecting the generated "entities" to work with JPA?

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.18.0, 3.17.5 (https://github.com/jOOQ/jOOQ/issues/14134), and 3.16.11 (https://github.com/jOOQ/jOOQ/issues/14135)

**Comment by rayman245** [comment]
> While I'm fixing this, can you explain why you're using jOOQ's `<jpaAnnotations/>` flag? You're expecting the generated "entities" to work with JPA?

Hi @lukaseder yes, we are using spring jpa to perform CRUD functionalities. This was working fine in Spring Boot 2, but broken in Spring Boot 3 due to the changes in Hibernate 6.

Will it be possible to keep the precision but just map it to Hibernate's implementation? Because we are using it in a batch application where we would like to dynamically let jpa update schema in the target DB using the pojos generated from jooq from source DB.

**Comment by lukaseder** [comment]
> Will it be possible to keep the precision but just map it to Hibernate's implementation? Because we are using it in a batch application where we would like to dynamically let jpa update schema in the target DB using the pojos generated from jooq from source DB.

I'm sorry, I don't understand what this means. The `precision` was generated because H2 reported it, but we don't really care about it and neither does Hibernate.

So, it might indeed be seen as a bug in jOOQ, which manifested via this incompatible change in Hibernate.

Why do you want to "keep" it? What does it mean to "map it to Hibernate's implementation"?

**Comment by rayman245** [comment]
> Just to be sure. Here you say:
> 
> > Hi @lukaseder I am getting this error when trying to generate DDL from Entity, only occurring for Double values with precision 53.
> 
> But on https://hibernate.atlassian.net/browse/HHH-15623, you say:
> 
> > As shown here the generated entity (From JOOQ) has the correct precision 53,image
> 
> What is it, then? Are you generating entities using jOOQ or are you using jOOQ to generate code from entities?

We are generating only the pojos, not the entities.

**Comment by lukaseder** [comment]
> We are generating only the pojos, not the entities.

Yes, but you're using them as entities? They're not...

**Comment by rayman245** [comment]
> > Will it be possible to keep the precision but just map it to Hibernate's implementation? Because we are using it in a batch application where we would like to dynamically let jpa update schema in the target DB using the pojos generated from jooq from source DB.
> 
> I'm sorry, I don't understand what this means. The `precision` was generated because H2 reported it, but we don't really care about it and neither does Hibernate.
> 
> So, it might indeed be seen as a bug in jOOQ, which manifested via this incompatible change in Hibernate.
> 
> Why do you want to "keep" it? What does it mean to "map it to Hibernate's implementation"?

I'll need to check back again, but if I am not mistaken this way JPA would be able to run the DDL but it will fail when trying to insert data which has decimal points.

Ill provide my test result by tomorrow on this.

**Comment by lukaseder** [comment]
> I'll need to check back again, but if I am not mistaken this way JPA would be able to run the DDL but it will fail when trying to insert data which has decimal points.

I'm quite confused now :) Anyway. The SQL `DOUBLE PRECISION` and `REAL` types don't actually have any precision that can be specified. `FLOAT` does, but not all RDBMS implement things this way. I don't see how removing the `precision` attribute would affect any of your logic. But then again, I don't really understand what you're doing.

I agree with Christian Beikov's comment on https://hibernate.atlassian.net/browse/HHH-15623 that you should use Flyway or Liquibase for proper database change management

**Comment by rayman245** [comment]
> The SQL `DOUBLE PRECISION` and `REAL` types don't actually have any precision that can be specified. `FLOAT` does, but not all RDBMS implement things this way. I don't see how removing the `precision` attribute would affect any of your logic. But then again, I don't really understand what you're doing.

Apologies on the confusion, what I meant by the above was I tried following Christian Beikov' solution:- After jooq generated the pojos, we manually removed the precision from the affected fields in the pojos and ran the build again this time with jooq generation off. JPA successfully created the tables in the target DB but when trying to perform insert to target DB(postgres), it threw an SQL exception regarding datatype with numbers which had decimal points. But I'm realizing this could have just been a fluke due to another unrelated development issue...

Thanks for your explaination above it helped me understand better.

But yeah... will check on the feasibility of liquibase and flyway for my case. 

Thanks again for the support @lukaseder.
